esx_service/cli/vmdkops_post_update.11.1.py

```diff
@@ -0,0 +1,192 @@
+    backup = dbfile + ".bck"
+    if os.path.isfile(backup):
+        print("Backup file '{0}' already exists - skipping DB backup".format(backup))
+    else:
+        print("Backing up Config DB to '{0}'".format(backup))
+        shutil.copy(dbfile, backup)
+
+    if stop_service:
+        print("Stopping vmdk-opsd service")
+        os.system("/etc/init.d/vmdk-opsd stop")
+
+    # STEP : convert dir names to new UUID if needed.
+    print("Starting conversion of _DEFAULT tenant directory names. old_uid is {0}".format(old_uuid))
+    stores = vmdk_utils.get_datastores()
+    if not stores:
+        print("Docker volume storage is not initialized - skipping directories patching")
+    else:
+        for ds_name, ds_id, ds_path in stores:
+            print("Working on Datastore '{0}'".format(ds_name))
+            # move stuff from old_uuid to new_uuid ()
+            new_dir = os.path.join(ds_path, STATIC_UUID)
+            old_dir = os.path.join(ds_path, old_uuid)
+            symlink_name = os.path.join(ds_path, STATIC_NAME)
+            print("  Info: ", new_dir, old_dir, symlink_name)
+            if not os.path.isdir(old_dir):
+                print("  Skipping {0} - not found".format(old_dir))
+                continue
+            if os.path.isdir(new_dir):
+                print("  Skipping {0} - {1} already exists".format(old_dir, new_dir))
+            print("  Renaming {0} to {1}".format(old_dir, new_dir))
+            os.rename(old_dir, new_dir)
+            try:
+                os.remove(symlink_name)
+            except:
+                pass
+            # TBD: make a short symlink here (not full path - 'cd' first)
+            os.symlink(new_dir, symlink_name)
+
+
+    # STEP: patch database
+    print("Working on DB patch...")
+
+    # sql for update the DB
+    # note that:
+    #       {0} is old_uuid (default tenant uuid pre-upgrade)
+    #       {1} is new_uuid (default tenant uuid post-upgrade)
+    #       {2} is tmp name - we need it to comply with DB constrants
+    #       {3} is default tenant description (from DB)
+    #       {4} is default DB for default tenant (from DB)
+    #       {5} is the name ("_DEFAULT") for default tenant
+    # TBD - use named params in formatting
+    sql_query_template = \
+    """
+    -- insert temp record to make foreign key happy
+    INSERT INTO tenants VALUES ( '{1}', '{2}', '{3}', '{4}' ) ;
+
+    -- update the tables
+    UPDATE vms SET tenant_id = '{1}' WHERE tenant_id = '{0}';
+    UPDATE volumes SET tenant_id = '{1}' WHERE tenant_id = '{0}';
+    UPDATE privileges SET tenant_id = '{1}' WHERE tenant_id = '{0}';
+
+    -- recover _DEFAULT tenant record
+    DELETE FROM tenants WHERE id = '{0}';
+    UPDATE tenants SET name = '{5}' WHERE name = '{2}';
+    UPDATE versions SET major_ver=1, minor_ver=1;
+    """
+
+    tmp_tenant_name = "__tmp_name_upgrade_0_11"
+    sql_query = sql_query_template.format(old_uuid, STATIC_UUID, tmp_tenant_name,
```

Comment:
How is it different from printing exception and exiting , which is the current behavior in case there is an exception ?